src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,39 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int STORAGE_SIZE = 10;
+    private K[] key;
+    private V[] value;
+    private int size;
+
+    public StorageImpl() {
+        key = (K[]) new Object[STORAGE_SIZE];
+        value = (V[]) new Object[STORAGE_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+        if (get(key) == null) {
+            this.key[size] = key;
+            this.value[size] = value;
+            size++;
```

Comment:
```suggestion
            key[size] = key;
            value[size] = value;
            size++;
```
according to last comment, you don't need `this` keyword at all